roofit/roofitcore/src/RooHistPdf.cxx

```diff
@@ -379,6 +399,24 @@ double RooHistPdf::analyticalIntegral(Int_t code,
   return ret ;
 }
 
+std::string RooHistPdf::rooHistIntegralTranslateImpl(int code, RooAbsArg const *klass, RooDataHist const *dataHist,
+                                                     const RooArgSet &obs)
+{
+   if (((2 << obs.size()) - 1) != code) {
+      oocoutE(klass, InputArguments)
+         << "RooHistPdf::integral(" << klass->GetName()
+         << ") ERROR: AD currently only supports integrating over all histogram observables." << std::endl;
+      return "";
+   }
+   return std::to_string(dataHist->sum(false));
+}
```

Comment:
```suggestion
std::string RooHistPdf::rooHistIntegralTranslateImpl(int code, RooAbsArg const *klass, RooDataHist const *dataHist,
                                                     const RooArgSet &obs, bool histFuncMode)
{
   if (((2 << obs.size()) - 1) != code) {
      oocoutE(klass, InputArguments)
         << "RooHistPdf::integral(" << klass->GetName()
         << ") ERROR: AD currently only supports integrating over all histogram observables." << std::endl;
      return "";
   }
   return std::to_string(dataHist->sum(histFuncMode));
}
```

If you look at the implementation of `RooHistPdf::analyticalIntegral()`, there is a flag for the RooHistFunc mode, which gets propagated to the `dataHist->sum()` call. You need to do the same here to get the correct results.

Also, why do you call the RooAbsArg to translate `klass`? That's a funny name that I have only seen for objects that represent classes, like `TClass` instances in C++, or any class in Python. Here, the object is not representing that, but a RooAbsArg instance. Why not just call it `arg` or better for clarity `translatedArg`? Also, it doesn't need to be a pointer because it can never be null. the same applies to the `dataHist` parameter.

Reply:
Honestly, I would rather have it named klass instead, as it refers to the class pointer for the class being translated. I just prefer to keep it that way as it makes more sense than the suggested names to me.